### PR TITLE
Application form feature

### PR DESCRIPTION
**Pull request by cloverzer0**
This pull request has the following:

- Added a new mutation procedure called submitApplication in the postRouter router. File path: src/server/api/routers/post.ts 
- Made an application form component ( expecting it to be changed ) to test if the right procedure occurs. File path: src/app/_components/ApplicationForm.tsx

Note: i made it possible to submit the form without filling all inputs the only necessary field is email. if i should change it to all let me know.

Screen Shot 2024-08-04 at 2 26 28 AM
Screen Shot 2024-08-04 at 2 26 59 AM
Screen Shot 2024-08-04 at 2 28 47 AM
Screen Shot 2024-08-04 at 2 31 22 AM

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import Link from 'next/link'
 import { UserProfile } from '~/app/_components/userProfile/UserProfile'
 import { getServerAuthSession } from '~/server/auth'
 import { HydrateClient, api } from '~/trpc/server'
+import ApplicationForm from "./_components/ApplicationForm";
```
Comment (MFarabi619) [inline src/app/page.tsx:6]: Refactor this to use path alias please, see similar feedback on Ryan's PR.

File: src/app/_components/ApplicationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        try {
+            await submitApplicationForm.mutateAsync(formData);
+            console.log('Form submitted successfully');
+          } catch (error) {
+            console.error('Error submitting form:', error);
+        }
+    }
+
+    return (
+        <div>
```
Comment (MFarabi619) [inline src/app/_components/ApplicationForm.tsx:44]: Is the wrapper div around the form necessary?

**Comment by MFarabi619** [comment]
@cloverzer0 Thanks for providing a description. I like that it has screenshots of the front-end. You don't have to attach screenshots of the code, that's what the code diffs panel is for. The code may also potentially change often, which will quickly make those code screenshots out of date, and even misleading.

Avoid putting the file paths too. Again, I can see those in the code diff.

Once our deployment is working, people can click on the preview link, see what the live previews look like, and interact with your feature. At that point, your screenshots should be more like a guide on how to get to that feature and use it.

I generally like to use this format:

| Before | After |
|:-:|:-:|
| image... | image...|

Once we have our visual test server up, you won't have to do that either. It can instead be more of a flow.

File: src/server/api/routers/post.ts [hunk 3]
```diff
@@ -39,4 +39,39 @@ export const postRouter = createTRPCRouter({
   getSecretMessage: protectedProcedure.query(() => {
     return 'you can now see this secret message!'
   }),
+
+  submitApplicationForm: publicProcedure
```
Comment (HasithDeAlwis) [inline src/server/api/routers/post.ts:43]: This is my mistake, could you make this a `protectedProcedure`?

**Review by HasithDeAlwis — COMMENTED** [review]
Overall nice work, thanks for learning the basics of tRPC so quickly and getting this done. Awesome stuff!

**Pull request closed without merging**